src/systemcmds/mtd/mtd.cpp

```diff
@@ -112,7 +113,7 @@ static int mtd_status(void)
 						ret = instances[i].part_dev[p]->ioctl(instances[i].part_dev[p], MTDIOC_GEOMETRY, (unsigned long)((uintptr_t)&geo));
 						printf("    partition: %u:\n", p);
 						printf("     name:   %s\n", instances[i].partition_names[p]);
-						printf("     blocks: %u (%u bytes)\n", geo.neraseblocks, erasesize * geo.neraseblocks);
+						printf("     blocks: %" PRIu32 " (%" PRIu32 " bytes)\n", geo.neraseblocks, erasesize * geo.neraseblocks);
```

Comment:
I think the latter PRIu32 here is wrong, or then this requires typecast. "erasesize" is unsigned long and "geo.neraseblocks" is uint32_t. The multiplication result will be "unsigned long" on platforms which have 64bit long, and this will fail (e.g. on riscv64-unknown-elf for me)

Reply:
@jlaitine - incorporated! Thank you.